# Error messages should exist in a JSON file that's shared across the workspace when using Storm Stack (`storm-stack/format-error-codes`)

ğŸ’¼âš ï¸ This rule is enabled in the following configs: ğŸŒŸ `recommended`, ğŸ”’
`strict`. This rule _warns_ in the ğŸ“‹ `base` config.

ğŸ”§ This rule is automatically fixable by the
[`--fix` CLI option](https://eslint.org/docs/latest/user-guide/command-line-interface#--fix).

<!-- end auto-generated rule header -->

## Rule Details

ğŸ‘ Examples of **incorrect** code for this rule:

```ts
const type = "GET";

throw new StormError(`Failed to process ${type} request`);
```

ğŸ‘ Examples of **correct** code for this rule:

```ts
const type = "GET";

throw new StormError({ type: "general", code: 123, args: { type } });
```

## Version

0.4.0
