# Disallow implicit global variables (`storm-stack/no-implicit-globals`)

ğŸ’¼âš ï¸ This rule is enabled in the following configs: ğŸŒŸ `recommended`, ğŸ”’
`strict`. This rule _warns_ in the ğŸ“‹ `base` config.

<!-- end auto-generated rule header -->

## Rule Details

ğŸ‘ Examples of **incorrect** code for this rule:

```js
const foo = 1;
```

ğŸ‘ Examples of **correct** code for this rule:

```js
(function () {
  const foo = 1;
})();
```

## Version

0.4.0
