# Disallow implicit global variables (`storm-stack/no-implicit-globals`)

üíº‚ö†Ô∏è This rule is enabled in the following configs: üåü `recommended`, üîí
`strict`. This rule _warns_ in the üìã `base` config.

<!-- end auto-generated rule header -->

## Rule Details

üëé Examples of **incorrect** code for this rule:

```js
/* -------------------------------------------------------------------

                  ‚ö° Storm Software - Storm Stack

 This code was released as part of the Storm Stack project. Storm Stack
 is maintained by Storm Software under the Apache-2.0 license, and is
 free for commercial and private use. For more information, please visit
 our licensing page at https://stormsoftware.com/projects/storm-stack/license.

 Website:                  https://stormsoftware.com
 Repository:               https://github.com/storm-software/storm-stack
 Documentation:            https://stormsoftware.com/projects/storm-stack/docs
 Contact:                  https://stormsoftware.com/contact

 SPDX-License-Identifier:  Apache-2.0

 ------------------------------------------------------------------- */

const foo = 1
```

üëç Examples of **correct** code for this rule:

```js
/* -------------------------------------------------------------------

                  ‚ö° Storm Software - Storm Stack

 This code was released as part of the Storm Stack project. Storm Stack
 is maintained by Storm Software under the Apache-2.0 license, and is
 free for commercial and private use. For more information, please visit
 our licensing page at https://stormsoftware.com/projects/storm-stack/license.

 Website:                  https://stormsoftware.com
 Repository:               https://github.com/storm-software/storm-stack
 Documentation:            https://stormsoftware.com/projects/storm-stack/docs
 Contact:                  https://stormsoftware.com/contact

 SPDX-License-Identifier:  Apache-2.0

 ------------------------------------------------------------------- */

;(function () {
  const foo = 1
})()
```

## Version

0.4.0
