# Prefer usage of `StormJSON` class when using Storm Stack (`storm-stack/storm-json-only`)

ğŸ’¼ğŸš« This rule is enabled in the ğŸ”’ `strict` config. This rule is _disabled_ in
the following configs: ğŸ“‹ `base`, ğŸŒŸ `recommended`.

ğŸ”§ This rule is automatically fixable by the
[`--fix` CLI option](https://eslint.org/docs/latest/user-guide/command-line-interface#--fix).

<!-- end auto-generated rule header -->

## Rule Details

ğŸ‘ Examples of **incorrect** code for this rule:

```ts
const value = JSON.parse(strValue);
```

ğŸ‘ Examples of **correct** code for this rule:

```ts
const value = StormJSON.parse(strValue);
```

## Version

0.4.0
