# Require usage of StormJSON class (`storm-stack/storm-json-only`)

ğŸ’¼ This rule is enabled in the âœ… `base` config.

<!-- end auto-generated rule header -->

## Rule Details

ğŸ‘ Examples of **incorrect** code for this rule:

```js
const value = JSON.parse(strValue);
```

ğŸ‘ Examples of **correct** code for this rule:

```js
const value = StormJSON.parse(strValue);
```

## Version

4.3.2
