# Require usage of StormError class (`storm-stack/storm-errors-only`)

ğŸ’¼ This rule is enabled in the âœ… `base` config.

<!-- end auto-generated rule header -->

## Rule Details

ğŸ‘ Examples of **incorrect** code for this rule:

```js
throw new Error("Failed to process request");
```

ğŸ‘ Examples of **correct** code for this rule:

```js
throw new StormError("Failed to process request");
```

## Version

4.3.2
