# Prefer usage of `StormURL` class when using Storm Stack (`storm-stack/storm-urls-only`)

ğŸ’¼âš ï¸ğŸš« This rule is enabled in the ğŸ”’ `strict` config. This rule _warns_ in the
ğŸŒŸ `recommended` config. This rule is _disabled_ in the ğŸ“‹ `base` config.

ğŸ”§ This rule is automatically fixable by the
[`--fix` CLI option](https://eslint.org/docs/latest/user-guide/command-line-interface#--fix).

<!-- end auto-generated rule header -->

## Rule Details

ğŸ‘ Examples of **incorrect** code for this rule:

```ts
const url = new URL("https://example.com")
```

ğŸ‘ Examples of **correct** code for this rule:

```ts
const url = new StormURL("https://example.com")
```

## Version

0.4.0
