# Require usage of StormRequest class (`storm-stack/storm-requests-only`)

ğŸ’¼ This rule is enabled in the âœ… `base` config.

<!-- end auto-generated rule header -->

## Rule Details

ğŸ‘ Examples of **incorrect** code for this rule:

```js
function handler(request: Request) {
  // ...
}
```

ğŸ‘ Examples of **correct** code for this rule:

```js
function handler(request: StormRequest) {
  // ...
}
```

## Version

4.3.2
