# Require usage of StormURL class (`storm-stack/storm-urls-only`)

ğŸ’¼ This rule is enabled in the âœ… `base` config.

<!-- end auto-generated rule header -->

## Rule Details

ğŸ‘ Examples of **incorrect** code for this rule:

```js
const url = new URL("https://example.com");
```

ğŸ‘ Examples of **correct** code for this rule:

```js
const url = new StormURL("https://example.com");
```

## Version

4.3.2
