diff --git a/src/command-line/release/changelog.js b/src/command-line/release/changelog.js
index 5a23b02b641d28279dd3ffe0b3425876aca596a2..f4acaa0da70c6169b7d7369f25f4c0302c45adb4 100644
--- a/src/command-line/release/changelog.js
+++ b/src/command-line/release/changelog.js
@@ -256,7 +256,10 @@ function createAPI(overrideReleaseConfig) {
                     continue;
                 }
                 const dependentProjects = (projectsVersionData[project].dependentProjects || [])
-                    .map((dep) => {
+                  .map((dep) => {
+
+                    console.log(`[storm-stack-debug]: Project Name: ${project}, Project Current Version: ${projectsVersionData[project].currentVersion}, Project New Version: ${projectsVersionData[project].newVersion}, Dependency Name: ${dep.source}`)
+
                     return {
                         dependencyName: dep.source,
                         newVersion: projectsVersionData[dep.source].newVersion,
@@ -731,7 +734,7 @@ async function generateChangelogForWorkspace({ tree, args, projectGraph, nxRelea
      */
     if (interactive) {
         const tmpDir = (0, tmp_1.dirSync)().name;
-        const changelogPath = (0, path_1.joinPathFragments)(tmpDir, 
+        const changelogPath = (0, path_1.joinPathFragments)(tmpDir,
         // Include the tree path in the name so that it is easier to identify which changelog file is being edited
         `PREVIEW__${interpolatedTreePath.replace(/\//g, '_')}`);
         (0, node_fs_1.writeFileSync)(changelogPath, contents);
@@ -833,7 +836,7 @@ async function generateChangelogForProjects({ tree, args, changes, projectsVersi
          */
         if (interactive) {
             const tmpDir = (0, tmp_1.dirSync)().name;
-            const changelogPath = (0, path_1.joinPathFragments)(tmpDir, 
+            const changelogPath = (0, path_1.joinPathFragments)(tmpDir,
             // Include the tree path in the name so that it is easier to identify which changelog file is being edited
             `PREVIEW__${interpolatedTreePath.replace(/\//g, '_')}`);
             (0, node_fs_1.writeFileSync)(changelogPath, contents);
@@ -861,7 +864,7 @@ async function generateChangelogForProjects({ tree, args, changes, projectsVersi
                 changelogContents = contents;
             }
             tree.write(interpolatedTreePath, changelogContents);
-            (0, print_changes_1.printAndFlushChanges)(tree, !!dryRun, 3, false, shared_1.noDiffInChangelogMessage, 
+            (0, print_changes_1.printAndFlushChanges)(tree, !!dryRun, 3, false, shared_1.noDiffInChangelogMessage,
             // Only print the change for the current changelog file at this point
             (f) => f.path === interpolatedTreePath);
         }
