diff --git a/index.js b/index.js
index 76649fe2f7989c6c259a9c1a5d17b50744e0f963..d03a3939e3862b4ad46e0440cdc54dfce211b580 100644
--- a/index.js
+++ b/index.js
@@ -64,25 +64,21 @@ switch (platform) {
   case 'android':
     switch (arch) {
       case 'arm64':
-        localFileExisted = existsSync(join(__dirname, 'resolver.android-arm64.node'))
+        localFileExisted = false; // existsSync(join(__dirname, 'resolver.android-arm64.node'))
         try {
-          if (localFileExisted) {
-            nativeBinding = require('./resolver.android-arm64.node')
-          } else {
-            nativeBinding = require('@oxc-resolver/binding-android-arm64')
-          }
+
+          nativeBinding = require('@oxc-resolver/binding-android-arm64')
+
         } catch (e) {
           loadError = e
         }
         break
       case 'arm':
-        localFileExisted = existsSync(join(__dirname, 'resolver.android-arm-eabi.node'))
+        localFileExisted = false; // existsSync(join(__dirname, 'resolver.android-arm-eabi.node'))
         try {
-          if (localFileExisted) {
-            nativeBinding = require('./resolver.android-arm-eabi.node')
-          } else {
-            nativeBinding = require('@oxc-resolver/binding-android-arm-eabi')
-          }
+
+          nativeBinding = require('@oxc-resolver/binding-android-arm-eabi')
+
         } catch (e) {
           loadError = e
         }
@@ -94,43 +90,37 @@ switch (platform) {
   case 'win32':
     switch (arch) {
       case 'x64':
-        localFileExisted = existsSync(
+        localFileExisted = false; /* existsSync(
           join(__dirname, 'resolver.win32-x64-msvc.node')
-        )
+        )*/
         try {
-          if (localFileExisted) {
-            nativeBinding = require('./resolver.win32-x64-msvc.node')
-          } else {
-            nativeBinding = require('@oxc-resolver/binding-win32-x64-msvc')
-          }
+
+          nativeBinding = require('@oxc-resolver/binding-win32-x64-msvc')
+
         } catch (e) {
           loadError = e
         }
         break
       case 'ia32':
-        localFileExisted = existsSync(
+        localFileExisted = false; /*existsSync(
           join(__dirname, 'resolver.win32-ia32-msvc.node')
-        )
+        )*/
         try {
-          if (localFileExisted) {
-            nativeBinding = require('./resolver.win32-ia32-msvc.node')
-          } else {
-            nativeBinding = require('@oxc-resolver/binding-win32-ia32-msvc')
-          }
+
+          nativeBinding = require('@oxc-resolver/binding-win32-ia32-msvc')
+
         } catch (e) {
           loadError = e
         }
         break
       case 'arm64':
-        localFileExisted = existsSync(
+        localFileExisted = false; /*existsSync(
           join(__dirname, 'resolver.win32-arm64-msvc.node')
-        )
+        )*/
         try {
-          if (localFileExisted) {
-            nativeBinding = require('./resolver.win32-arm64-msvc.node')
-          } else {
-            nativeBinding = require('@oxc-resolver/binding-win32-arm64-msvc')
-          }
+
+          nativeBinding = require('@oxc-resolver/binding-win32-arm64-msvc')
+
         } catch (e) {
           loadError = e
         }
@@ -140,38 +130,32 @@ switch (platform) {
     }
     break
   case 'darwin':
-    localFileExisted = existsSync(join(__dirname, 'resolver.darwin-universal.node'))
+    localFileExisted = false; // existsSync(join(__dirname, 'resolver.darwin-universal.node'))
     try {
-      if (localFileExisted) {
-        nativeBinding = require('./resolver.darwin-universal.node')
-      } else {
-        nativeBinding = require('@oxc-resolver/binding-darwin-universal')
-      }
+
+      nativeBinding = require('@oxc-resolver/binding-darwin-universal')
+
       break
-    } catch {}
+    } catch { }
     switch (arch) {
       case 'x64':
-        localFileExisted = existsSync(join(__dirname, 'resolver.darwin-x64.node'))
+        localFileExisted = false; // existsSync(join(__dirname, 'resolver.darwin-x64.node'))
         try {
-          if (localFileExisted) {
-            nativeBinding = require('./resolver.darwin-x64.node')
-          } else {
-            nativeBinding = require('@oxc-resolver/binding-darwin-x64')
-          }
+
+          nativeBinding = require('@oxc-resolver/binding-darwin-x64')
+
         } catch (e) {
           loadError = e
         }
         break
       case 'arm64':
-        localFileExisted = existsSync(
+        localFileExisted = false; /*existsSync(
           join(__dirname, 'resolver.darwin-arm64.node')
-        )
+        )*/
         try {
-          if (localFileExisted) {
-            nativeBinding = require('./resolver.darwin-arm64.node')
-          } else {
-            nativeBinding = require('@oxc-resolver/binding-darwin-arm64')
-          }
+
+          nativeBinding = require('@oxc-resolver/binding-darwin-arm64')
+
         } catch (e) {
           loadError = e
         }
@@ -184,13 +168,11 @@ switch (platform) {
     if (arch !== 'x64') {
       throw new Error(`Unsupported architecture on FreeBSD: ${arch}`)
     }
-    localFileExisted = existsSync(join(__dirname, 'resolver.freebsd-x64.node'))
+    localFileExisted = false; // existsSync(join(__dirname, 'resolver.freebsd-x64.node'))
     try {
-      if (localFileExisted) {
-        nativeBinding = require('./resolver.freebsd-x64.node')
-      } else {
-        nativeBinding = require('@oxc-resolver/binding-freebsd-x64')
-      }
+
+      nativeBinding = require('@oxc-resolver/binding-freebsd-x64')
+
     } catch (e) {
       loadError = e
     }
@@ -199,28 +181,24 @@ switch (platform) {
     switch (arch) {
       case 'x64':
         if (isMusl()) {
-          localFileExisted = existsSync(
+          localFileExisted = false; /* existsSync(
             join(__dirname, 'resolver.linux-x64-musl.node')
-          )
+          )*/
           try {
-            if (localFileExisted) {
-              nativeBinding = require('./resolver.linux-x64-musl.node')
-            } else {
-              nativeBinding = require('@oxc-resolver/binding-linux-x64-musl')
-            }
+
+            nativeBinding = require('@oxc-resolver/binding-linux-x64-musl')
+
           } catch (e) {
             loadError = e
           }
         } else {
-          localFileExisted = existsSync(
+          localFileExisted = false; /* existsSync(
             join(__dirname, 'resolver.linux-x64-gnu.node')
-          )
+          )*/
           try {
-            if (localFileExisted) {
-              nativeBinding = require('./resolver.linux-x64-gnu.node')
-            } else {
-              nativeBinding = require('@oxc-resolver/binding-linux-x64-gnu')
-            }
+
+            nativeBinding = require('@oxc-resolver/binding-linux-x64-gnu')
+
           } catch (e) {
             loadError = e
           }
@@ -228,86 +206,74 @@ switch (platform) {
         break
       case 'arm64':
         if (isMusl()) {
-          localFileExisted = existsSync(
+          localFileExisted = false; /* existsSync(
             join(__dirname, 'resolver.linux-arm64-musl.node')
-          )
+          )*/
           try {
-            if (localFileExisted) {
-              nativeBinding = require('./resolver.linux-arm64-musl.node')
-            } else {
-              nativeBinding = require('@oxc-resolver/binding-linux-arm64-musl')
-            }
+
+            nativeBinding = require('@oxc-resolver/binding-linux-arm64-musl')
+
           } catch (e) {
             loadError = e
           }
         } else {
-          localFileExisted = existsSync(
+          localFileExisted = false; /* existsSync(
             join(__dirname, 'resolver.linux-arm64-gnu.node')
-          )
+          )*/
           try {
-            if (localFileExisted) {
-              nativeBinding = require('./resolver.linux-arm64-gnu.node')
-            } else {
-              nativeBinding = require('@oxc-resolver/binding-linux-arm64-gnu')
-            }
+
+            nativeBinding = require('@oxc-resolver/binding-linux-arm64-gnu')
+
           } catch (e) {
             loadError = e
           }
         }
         break
       case 'arm':
-        localFileExisted = existsSync(
+        localFileExisted = false; /*existsSync(
           join(__dirname, 'resolver.linux-arm-gnueabihf.node')
-        )
+        )*/
         try {
-          if (localFileExisted) {
-            nativeBinding = require('./resolver.linux-arm-gnueabihf.node')
-          } else {
-            nativeBinding = require('@oxc-resolver/binding-linux-arm-gnueabihf')
-          }
+
+          nativeBinding = require('@oxc-resolver/binding-linux-arm-gnueabihf')
+
         } catch (e) {
           loadError = e
         }
         break
       case 'riscv64':
         if (isMusl()) {
-          localFileExisted = existsSync(
+          localFileExisted = false; /*existsSync(
             join(__dirname, 'resolver.linux-riscv64-musl.node')
-          )
+          )*/
           try {
-            if (localFileExisted) {
-              nativeBinding = require('./resolver.linux-riscv64-musl.node')
-            } else {
-              nativeBinding = require('@oxc-resolver/binding-linux-riscv64-musl')
-            }
+
+            nativeBinding = require('@oxc-resolver/binding-linux-riscv64-musl')
+
           } catch (e) {
             loadError = e
           }
         } else {
-          localFileExisted = existsSync(
+          localFileExisted = false; /*existsSync(
             join(__dirname, 'resolver.linux-riscv64-gnu.node')
-          )
+          )*/
           try {
-            if (localFileExisted) {
-              nativeBinding = require('./resolver.linux-riscv64-gnu.node')
-            } else {
-              nativeBinding = require('@oxc-resolver/binding-linux-riscv64-gnu')
-            }
+
+            nativeBinding = require('@oxc-resolver/binding-linux-riscv64-gnu')
+
           } catch (e) {
             loadError = e
           }
         }
         break
       case 's390x':
-        localFileExisted = existsSync(
+        localFileExisted = false; /*existsSync(
           join(__dirname, 'resolver.linux-s390x-gnu.node')
-        )
+        )*/
         try {
-          if (localFileExisted) {
-            nativeBinding = require('./resolver.linux-s390x-gnu.node')
-          } else {
-            nativeBinding = require('@oxc-resolver/binding-linux-s390x-gnu')
-          }
+
+          nativeBinding = require('@oxc-resolver/binding-linux-s390x-gnu')
+
         } catch (e) {
           loadError = e
         }
