# Prefer usage of `StormError` class when using Storm Stack (`storm-stack/storm-errors-only`)

ğŸ’¼âš ï¸ This rule is enabled in the following configs: ğŸŒŸ `recommended`, ğŸ”’
`strict`. This rule _warns_ in the ğŸ“‹ `base` config.

ğŸ”§ This rule is automatically fixable by the
[`--fix` CLI option](https://eslint.org/docs/latest/user-guide/command-line-interface#--fix).

<!-- end auto-generated rule header -->

## Rule Details

ğŸ‘ Examples of **incorrect** code for this rule:

```ts
throw new Error("Failed to process request");
```

ğŸ‘ Examples of **correct** code for this rule:

```ts
throw new StormError("Failed to process request");
```

## Version

0.4.0
